### PR TITLE
RavenDB-9919 - Problem migration indexes from 3.5 automatically

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-9919

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs [hunk 1]
```diff
@@ -351,6 +338,78 @@ private SmugglerProgressBase.Counts ProcessIndexes(SmugglerResult result)
...
+                {
+                    errorMessage += Environment.NewLine + $"Reduce: {indexDefinition.Reduce}";
+                }
+                result.AddError(errorMessage);
+            }
+
+            string ReplaceLegacyMethods(string str)
+            {
+                return str
+                    .Replace("AbstractIndexCreationTask.SpatialGenerate", "CreateSpatialField")
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: We do not want to replace them, just write a proper error as we do for LoadDocument

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs [hunk 2]
```diff
@@ -351,6 +338,78 @@ private SmugglerProgressBase.Counts ProcessIndexes(SmugglerResult result)
...
+                    foreach (var indexDefinitionField in indexDefinition.Fields)
+                        indexDefinitionField.Value.Analyzer = null;
+                }
+
+                actions.WriteIndex(indexDefinition);
+            }
+            catch (Exception e)
+            {
+                var exceptionMessage = e.Message;
+                if (exceptionMessage.Contains("No overload for method 'LoadDocument' takes 1 arguments"))
```
Comment (ayende) [inline src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: You are relying on the _error message_ from the compiler.
But if you are using a machine with localized error messages, see:
https://github.com/dotnet/roslyn/blob/ac2da0f6985620a8dea9aca7c2a00848df3b4e8f/src/Compilers/CSharp/Portable/CSharpResources.resx#L666

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: fixed

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: done

**Comment by grisha-kotler** [comment]
Pushed my changes

**Pull request merged**